test/test_broker.hpp

```diff
@@ -997,11 +1017,26 @@ class test_broker {
             BOOST_ASSERT(non_active_sessions_.get<tag_client_id>().count(client_id) == 0);
             BOOST_ASSERT(non_active_sessions_.get<tag_client_id>().find(client_id) == non_active_sessions_.get<tag_client_id>().end());
 
+            auto const& sei_opt = state->session_expiry_interval;
+            if (sei_opt && sei_opt.value() != std::chrono::seconds(0xffffffffUL)) {
```

Comment:
std::numeric_limits<std::chrono::seconds::rep>::max()

Reply:
Better

```cpp
std::chrono::seconds::max()
```

Reply:
I think that std::chrono::seconds can hold the value that is greater than 0xffffffff.
It is limited by MQTT v5 spec. 0xffffffff has the special meaning that never expire.
I thin that I should define `static constexpr std::uint32_t const session_never_expire = 0xffffffffUL` somewhere.
What do you think?

Reply:
Ok.

Yes, this should be a static constexpr with documentation. I agree.

Yes, should not be ::max().